### PR TITLE
Support optional inclusion of `start`, `end` boundaries in  `load`

### LINKED ISSUE #23: Support optional inclusion of end boundary in load function
DJ processes data chunk by chunk and we are finding duplicates at the edge of chunks from the fact the end timestamp is included by default when loading the data through Aeon's api load function. In general this has been a shared experience for many of us when we wish to analyse data on a chunk or block level. The intuitive way to load a chunk is to set `start = hour` and `end = hour + 1`, but more often than not the final timestamp at `hour + 1`  is not of interest, requiring repetitive code get rid of it each time.

An `include_end` argument could be introduced to give users the choice as to whether or not the load function should return the end timestamp.

### PR DESCRIPTION
**Pull request by lochhh**
Datetime range querying is fundamental to aeon datasets, so here we introduce optional parameters for precisely specifying how to deal with range boundaries, similar to the [pandas.date_range](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html) and [pandas.DataFrame.between_time](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.between_time.html) functions, which also have an inclusive parameter.

Unit tests have been added to cover possible edge cases.

Closes #23

### CODE REVIEW

File: swc/aeon/io/api.py [hunk 1]
```diff
@@ -89,7 +89,39 @@ def _empty(columns):
...
+    elif inclusive == "neither":
+        result = result.iloc[1:] if first_idx_equals_start else result
+        result = result.iloc[:-1] if last_idx_equals_end else result
+        return result
+    else:
+        raise ValueError(f"Invalid inclusive: {inclusive!r}")
+
+
+def load(
+    root, reader, start=None, end=None, inclusive="both", time=None, tolerance=None, epoch=None, **kwargs
```
[the remark this answers is not in the document]
Comment (glopesdev) [reply swc/aeon/io/api.py:123]: I also wonder, if we go this route,  whether we should revisit the motivation for the `epoch` parameter.

The problem with having all these parameters is it becomes confusing to document and use how different combinations of them would be interpreted together, e.g. what would `epoch = hour, inclusive = "end"` mean? Should it throw an error? Be ignored? Etc.

Maybe if there is a clear way to specify chunk level searches this way we could cleanup the api somehow.

[the remark this answers is not in the document]
Comment (lochhh) [reply swc/aeon/io/api.py:123]: I've tried to clarify the docstrings for `inclusive`, adding:  This is only applicable when both ``start`` and ``end`` are provided.

File: swc/aeon/io/api.py [hunk 2]
```diff
@@ -100,6 +132,9 @@ def load(root, reader, start=None, end=None, time=None, tolerance=None, epoch=No
     :param Reader reader: A data stream reader object used to read chunk data from the dataset.
     :param datetime, optional start: The left bound of the time range to extract.
     :param datetime, optional end: The right bound of the time range to extract.
+    :param str, optional inclusive: Specifies whether the ``start`` and ``end`` bounds are inclusive or
+        exclusive. This is only applicable when both ``start`` and ``end`` are provided.
```
Comment (glopesdev) [inline swc/aeon/io/api.py]:
```suggestion
        exclusive. This is only applicable when ``start`` and/or ``end`` are provided. 
```

Reading the code it looks like filter gets called when any one of them is defined. Not sure if we have tests covering all those edge cases, but assuming this is true I would suggest rewording as above?
